packages/layer-composer/src/generators/tile-cluster/tile-cluster.ts

```diff
@@ -0,0 +1,130 @@
+import { AnyLayer } from '@globalfishingwatch/mapbox-gl'
+import { Type, TileClusterGeneratorConfig, MergedGeneratorConfig } from '../types'
+import { isUrlAbsolute } from '../../utils'
+import { isConfigVisible } from '../utils'
+import { API_GATEWAY } from '../../layer-composer'
+import { Group } from '../..'
+
+export type GlobalTileClusterGeneratorConfig = Required<
+  MergedGeneratorConfig<TileClusterGeneratorConfig>
+>
+
+class TileClusterGenerator {
+  type = Type.TileCluster
+
+  _getStyleSources = (config: GlobalTileClusterGeneratorConfig) => {
+    if (!config.tilesUrl) {
+      throw new Error(`Tile Cluster layer should specify tilesUrl ${JSON.stringify(config)}`)
+    }
+    const tilesUrl = isUrlAbsolute(config.tilesUrl)
+      ? config.tilesUrl
+      : API_GATEWAY + config.tilesUrl
+    const url = new URL(tilesUrl.replace(/{{/g, '{').replace(/}}/g, '}'))
```

Comment:
Could this be fixed in the API instead?

Reply:
Noted and I will open the conversation in the next meeting